File: src/main/java/Programers/옹알이.java
package Programers;

import java.util.Arrays;
import java.util.List;

public class 옹알이 {
    private final List<String> AVAIL_BABBLING = List.of("aya", "ye", "woo", "ma");

    public int solution(String[] babbling) {
        return Math.toIntExact(Arrays.stream(babbling).filter(this::canBabbling).count());
    }

    private boolean canBabbling(String babbling) {
        for(String b: AVAIL_BABBLING){
            if(babbling.contains(b+b)) {
                return false;
            }
        }
        for(String b: AVAIL_BABBLING) {
            babbling = babbling.replaceAll(b, "-");
        }

        return babbling.replaceAll("-", "").isEmpty();
    }

    public static void main(String[] args) {
        옹알이 sol = new 옹알이();
        System.out.println("@@@@@@@@@ :: "+sol.solution(new String[]{"ayaye", "uuu", "yeye", "yemawoo", "ayaayaa"}));
    }
}

File: src/main/java/Programers/kotlyn/옹알이.kt
package Programers.kotlyn

import kotlin.collections.count
import kotlin.text.contains
import kotlin.text.isEmpty
import kotlin.text.replace

class 옹알이 {
    private val AVAIL_BABBLING = listOf("aya", "ye", "woo", "ma")

    fun solution(babbling: Array<String>): Int {
        return babbling.count { canBabbling(it) }
    }

    private fun canBabbling(s0: String): Boolean {
        var s = s0

        for (b in AVAIL_BABBLING) {
            if (s.contains(b + b)) return false
        }

        for (b in AVAIL_BABBLING) {
            s = s.replace(b, "-")
        }

        return s.replace("-", "").isEmpty()
    }
}